### ISSUE TITLE
FES integration to retrieve Client Configuration (OrgRules)

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#276]
part of #275 

1) steps 1-5 must be done first  from issue https://github.com/FlowCrypt/flowcrypt-ios/issues/339
7) remove toast message from step 5 from issue https://github.com/FlowCrypt/flowcrypt-ios/issues/339
8) instead of showing toast, call `GET <fes>/api/v1/client-configuration?domain=example.com` (no authentication). The response is `{ clientConfiguration: DomainRulesJson }`. If this step was not skipped (because FES was not running at all earlier when testing `/api/` and this call fails, show error and don't let user proceed unless they can successfully retry this call.
9) Save this OrgRules json into storage, separately for each account. If it was in storage for this account before, blindly overwrite the previous OrgRules.

Here's an example: https://fes.flowcrypt.com/api/v1/client-configuration?domain=flowcrypt.com

**Comment by tomholub** [comment]
Also add the following class ported to Swift. The class won't actually be used anywhere yet (use of this class in the rest of the app will be done in a followup issue). https://github.com/FlowCrypt/flowcrypt-browser/blob/master/extension/js/common/org-rules.ts

**Comment by tomholub** [comment]
Let's do this after #310

**Comment by ekievsky** [comment]
@tomholub is it good to go?

**Comment by tomholub** [comment]
Yes, but it has to happen immediately after authentication, doing it in background in inbox is too late in terms of the first setup because we will use the information from OrgRules during setup procedure.

However, after you've done this during setup, it's ok to do the same in background everytime inbox is loaded, as that will take care of ongoing refresh (even if it means that it gets pulled twice unnecessarily during first setup - once after authentication and once when user gets to inbox - that's fine).

If unsure please ask. Thanks!

**Comment by tomholub** [comment]
What is the best way to save this? On other platforms we just save it as a whole json string (I think) but I'm flexible.

**Comment by ekievsky** [comment]
We may save it as parsed OrgRules model in UserDefaults

**Comment by tomholub** [comment]
That's ok. Would that mean we have to do a database migration every time the structure of OrgRules is extended? Like, when a new property is added to it that wasn't there before.

**Comment by ekievsky** [comment]
The database is going to store DomainRules which has following fields:
```
    let flags: [DomainRulesFlag]?
    let customKeyserverUrl: String?
    let keyManagerUrl: String?
    let disallowAttesterSearchForDomains: [String]?
    let enforceKeygenAlgo: String?
    let enforceKeygenExpireMonths: Int?
```

**Comment by tomholub** [comment]
Ok. That means if we add another field later, maybe Realm will have to do a schema migration. Maybe this type of simple migration (adding a nullable column) is done automatically by Realm?

**Comment by ekievsky** [comment]
We will just need to increase schema version.

**Comment by tomholub** [comment]
ok
